src/metabase/db/custom_migrations.clj

```diff
@@ -1070,6 +1070,20 @@
                                            :from   [:revision]
                                            :where  [:= :model "Card"]})))))
 
+(define-migration DeleteScanFieldValuesTriggerForDBThatTurnItOff
+  (let [dbs-without-scan-field-values (filter #(and (-> % :details :let-user-control-scheduling)
+                                                    (false? (:is_full_sync %)))
+                                              (t2/select :model/Database))]
+    (when (seq dbs-without-scan-field-values)
+      (classloader/the-classloader)
+      (set-jdbc-backend-properties!)
+      (let [scheduler (qs/initialize)]
+        (qs/start scheduler)
+        (doseq [db dbs-without-scan-field-values]
+          (qs/delete-trigger scheduler (triggers/key (format "metabase.task.update-field-values.trigger.%d" (:id db)))))
+        (t2/update! :model/Database :id [:in (map :id dbs-without-scan-field-values)] {:cache_field_values_schedule nil})
```

Comment:
It looks like we only ever use the ids from the database, so we might as well not select any other fields, and immediately map to just these values.

Reply:
we also need details, is_full_sync, is_on_demand for the filter.